keyboards/planck/keymaps/purple-rw/keymap.c

```diff
@@ -0,0 +1,548 @@
+/* Copyright 2020-2023 Rhys Wong
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * Only 34 keys (3x5+2 each hand) are really needed.
+ * Additional keys are for conveninence, but limited by available keycaps.
+ * Designed for use with Linux in mind, with US/English layout.
+ */
+
+#include QMK_KEYBOARD_H
+#ifdef AUDIO_ENABLE
+#include "muse.h"
+#endif
+
+
+enum planck_layers {
+    _COLEMAK,	// Colemak-DH
+    _QWERTY,    // Qwerty
+    _HANDSDN,   // Hands Down Neu
+    _SEMIMAK,   // Semimak JQ
+    _SYMBNUM,   // Numbers & Symbols
+    _SYMBL,     // Symbols
+    _NAVFUNC,   // Navigation & Functions
+    _PADFUNC,   // Keypad & Functions2
+    _MODS,      // Modifiers & Multimedia
+    _MOUSE,     // Mouse
+    _MACRO,     // Dynamic Macros
+    _PLOVER,    // Plover
+    _ADJST      // Adjustment
+};
+
+enum planck_keycodes {
+    COLEMAK = SAFE_RANGE,
+    QWERTY,
+    HANDSDN,
+    SEMIMAK,
+    PLOVER,
+    EXT_PLV
+};
+
+// Thumb keys
+#define RAISE   TT(_SYMBNUM)
+#define SYMBL   TT(_SYMBL)
+#define LOWER   TT(_NAVFUNC)
+#define FUNC2   TT(_PADFUNC)
+#define MODS    TT(_MODS)
+
+// Layer keys
+#define MOUSE   TG(_MOUSE)
+#define ADJST   TG(_ADJST)
+#define MACRO   OSL(_MACRO)
+
+// One-shot modifier keys
+#define OS_LSFT OSM(MOD_LSFT)
+#define OS_LCTL OSM(MOD_LCTL)
+#define OS_LALT OSM(MOD_LALT)
+#define OS_LGUI OSM(MOD_LGUI)
+#define OS_RSFT OSM(MOD_RSFT)
+#define OS_RCTL OSM(MOD_RCTL)
+#define OS_RALT OSM(MOD_RALT)
+#define OS_RGUI OSM(MOD_RGUI)
+
+// Equivalent layer names
+#define _RAISE  _SYMBNUM
+#define _LOWER  _NAVFUNC
```

Comment:
There aliases look kinda useless as of now.

You are using these aliases **only** at some points (eg: tri-layer) but not on other (eg: using `TT(_SYMBNUM)` instead of `TT(_RAISE)`)

Thus, your code would almost for sure break if you -for some reason- change the `_RAISE`/`_LOWER` aliases to point at any other layers. I would not use these aliases as they seem useless (or use them everywhere if your really want them)